Split up `usage_example` into 3 notebooks

mezarque opened the pull request:
<!--
Many thanks for contributing to Arcadia-Science/arcadia-pycolor!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).
-->

## PR checklist

- [x] Describe the changes you've made.
- [x] Describe any tests you have conducted to confirm that your changes behave as expected.

## Overview

This PR splits the `usage_example.ipynb` into three documents:
- `color_usage.ipynb` covering the color, palette, and gradient-associated parts of the API.
- `style_usage.ipynb` covering the `matplotlib` / `seaborn` -altering portions of the API under the `mpl` module.
- `cvd_usage.ipynb` covering the `cvd` utilities.

## Testing

All of the notebooks run just fine; would love feedback on copy and any other examples that would be useful to include in this documentation!

## Note

I'd wait to review this PR until #29 is merged, since I merged those changes into the branch I was working on for this.

keithchev commented on color_usage.ipynb:
```diff
@@ -0,0 +1,944 @@
+{
```
<p>Small bug: this cell doesn't run because the <code>all</code> attribute was removed in the prior PR.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Arcadia-Science/arcadia-pycolor/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

mezarque replied:
<p>Thanks for catching this!</p>

keithchev commented on style_usage.ipynb:
```diff
@@ -1,321 +1,20 @@
 {
```
<p>I'd add two things: </p><ul><li>mention that Suisse fonts are the "official" fonts specified by arcadia's styleguide </li><li>mention that if Suisse fonts aren't installed, <code>apc.mpl.setup</code> will still work, but matplotlib will fall back to using the user's existing default fonts. </li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Arcadia-Science/arcadia-pycolor/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

keithchev replied:
<p>These changes feel more important (to me) than what is implied by the word "polishing". I'd favor language that's more specific and motivating. Maybe "Generating publication-ready plots"? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Arcadia-Science/arcadia-pycolor/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

keithchev replied:
<p>Nice!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Arcadia-Science/arcadia-pycolor/pull/30/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

keithchev commented:
This looks good! I made some inline comments via nbreview. General comments:
- I don't think the sections in the notebook need to be numbered, since they aren't dependent on one another and people could skip around and run the cells in any order (for the most part).
- I think we'll want to move these notebooks into a `docs/` or `notebooks/` subdirectory, but this can wait for a later PR (maybe I can do this when I add the quickstart guide, which for now will be another notebook).

mezarque commented on color_usage.ipynb:
```diff
@@ -0,0 +1,944 @@
+{
```
<p>Yeah, that's what I'm trying to communicate. Maybe I can just state that more explicitly in the preceding text.</p>

mezarque commented on color_usage.ipynb:
```diff
@@ -0,0 +1,944 @@
+{
```
<p>Added an example!</p>

mezarque commented:
<p>Ok, I addressed your comments! I'll let you handle the re-arrangement of things. I've also left the paths for saving the PDF files in place for now; probably we want to put them into a folder that's then added to the <code>.gitignore</code> to avoid committing those PDFs in the future. </p><p> </p><p>In the <code>style_usage.ipynb</code> I also added a section that explicitly explains that the colors / gradients are registered to <code>matplotlib</code> and accessible by name, as I think it's a nice convenience feature that wasn't clearly explained before.  </p>

keithchev reviewed (approved):
LGTM! It was a good catch to mention the registered colormaps; I totally missed this.
